addons/l10n_th/views/report_invoice.xml

```diff
@@ -0,0 +1,97 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="debit_note_document" inherit_id="account.report_invoice_document" primary="True">
+        <xpath expr="//div[@name='address_not_same_as_shipping']//span[@t-field='o.partner_id.vat']" position="after">
+            <t t-if="o.company_id.account_fiscal_country_id.code == 'TH'">
+                <span t-esc="o.partner_id._l10n_th_get_branch_name()"/>
```

Comment:
Not really fan of calling function in template. What about having a compute non stored?